apps/sertra/components/mod-selector/search.tsx

```diff
@@ -18,6 +18,70 @@ export const SearchBox = <T,>({
 }: SearchBoxProps<T>) => {
   const [isFocused, setFocused] = useState<boolean>(false);
 
+  function focusFirstOption(target: EventTarget & HTMLInputElement) {
+    if (
+      target.parentElement?.parentElement?.nextElementSibling?.children[0]
+        ?.children[0]
+    ) {
+      target.parentElement.parentElement.nextElementSibling.children[0].children[0].focus();
+    }
+  }
+
+  function focusSearch(target: EventTarget & HTMLButtonElement) {
+    if (
+      target.parentElement?.parentElement?.previousElementSibling?.children[1]
+        ?.children[0]
+    ) {
+      target.parentElement.parentElement.previousElementSibling.children[1].children[0].focus();
+    }
+  }
+
+  const searchKeyDownHandler = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.code === "Escape") {
+      setFocused(false);
+    } else {
+      if (e.code === "ArrowDown") {
+        e.preventDefault();
+        if (e.currentTarget) {
+          focusFirstOption(e.currentTarget);
+        }
+      }
+    }
+  };
+
+  const listKeyDownHandler = (
```

Comment:
Depending on the amount of keyboard stuff we're going to implement, we might want to consider a library for this to make the code more readable/maintainable. [Mousetrap](https://github.com/ccampbell/mousetrap) has worked for me in the past, but I'm not sure if it's still a relevant option today.

Anyway, that's probably outside the scope of this task. Consider adding a task to ClickUp anyway.

Reply:
Added a task for this.